File: benchmark/c/cpw/find_index_given_fibonacci_number_constant_time/find_index_given_fibonacci_number_constant_time.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int find_index_given_fibonacci_number_constant_time ( int n ) ;
int find_index_given_fibonacci_number_constant_time ( int n ) {
  if ( n <= 1 ) return n;
  int a = 0, b = 1, c = 1;
  int res = 1;
  while ( c < n ) {
    c = a + b;
    res ++;
    a = b;
    b = c;
  }
  return res;
}

File: translation_result/else_bms/fail/c/cpw/find_index_given_fibonacci_number_constant_time/gemini/Restart/find_index_given_fibonacci_number_constant_time.rs

fn find_index_given_fibonacci_number_constant_time(n: i32) -> i32 {
    if n <= 1 {
        return n;
    }
    let mut a = 0;
    let mut b = 1;
    let mut c = 1;
    let mut res = 1;

    while c < n {
        c = a + b;
        res += 1;
        a = b;
        b = c;
    }

    res
}
